python/dgl/nn/pytorch/graph_transformer.py

```diff
@@ -216,6 +240,138 @@ def forward(self, ndata, attn_bias=None, attn_mask=None):
 
         attn = th.bmm(attn_weights, v_h).transpose(0, 1)
 
-        attn = self.out_proj(attn.reshape(N, bsz, self.feat_size).transpose(0, 1))
+        attn = self.out_proj(
+            attn.reshape(N, bsz, self.feat_size).transpose(0, 1)
+        )
 
         return attn
+
+
+class SparseBiasedMultiheadAttention(nn.Module):
+    r"""Sparse Multi-Head Attention Module with Graph Attention Bias.
+
+    Implement sparse attention computation restricted to neighborhoods of
+    nodes, as introduced in `A Generalization of Transformer Networks to
+    Graphs <https://arxiv.org/pdf/2012.09699>`__
+
+    .. math::
+
+        \text{Attn}_{ij}=\text{softmax}_{j\in\mathcal{N}(i)}(\dfrac{Q_iK_j^T}
+        {\sqrt{d}} \circ b)
+
+    :math:`Q_i` and :math:`K_j` are feature representation of nodes. :math:`d`
+    is the corresponding :attr:`feat_size`. :math:`b` is attention bias, which
+    can be additive or multiplicative according to the operator :math:`\circ`.
+    :math:`\mathcal{N}(i)` is the set of source nodes that have an edge to the
+    destination node :math:`i`.
+
+    Parameters
+    ----------
+    feat_size : int
+        Feature size.
+    num_heads : int
+        Number of attention heads, by which attr:`feat_size` is divisible.
+    bias : bool, optional
+        If True, it uses bias for linear projection. Default: True.
+    attn_bias_type : str, optional
+        The type of attention bias used for modifying attention. Selected from
+        'add' or 'mul'. Default: 'add'.
+
+        * 'add' is for additive attention bias.
+        * 'mul' is for multiplicative attention bias.
+    attn_drop : float, optional
+        Dropout probability on attention weights. Defalt: 0.
+
+    Examples
+    --------
+    >>> import dgl
+    >>> import torch as th
+    >>> from dgl.nn import SparseBiasedMultiheadAttention
+

```

Comment:
Maybe first declare `num_nodes = 100` and `num_edges = 80` to make the code more readable.